decode_test.go

```diff
@@ -1156,7 +1156,8 @@ func TestParserErrorUnmarshal(t *testing.T) {
 	expectedErr := &yaml.ParserError{
 		Message: "could not find expected ':'",
 		Line:    2,
-		Column:  0,
+		Column:  1,
```

Comment:
Also I feel like you are fixing the column computing as you said.

I'm a bit afraid it brings confusion with the PR.

Do you think you could open two PRs?

One for fixing line and column that were already there. And one to add the index.

But maybe you and other maintainers would find it over kills

Reply:
What are your thoughts who this ?

@itchyny @ingydotnet

Reply:
Separated into #135.